sdk/keyvault/azure-security-keyvault-certificates/src/test/java/com/azure/security/keyvault/certificates/CertificateClientTestBase.java

```diff
@@ -613,14 +613,14 @@ public void sleep(long millis) {
     static Stream<Arguments> getTestParameters() {
         // when this issues is closed, the newer version of junit will have better support for
         // cartesian product of arguments - https://github.com/junit-team/junit5/issues/1427
-        List<Arguments> argumentsList = new ArrayList<>();
-        getHttpClients()
-            .forEach(httpClient -> {
-                Arrays.stream(CertificateServiceVersion.values()).filter(
-                    CertificateClientTestBase::shouldServiceVersionBeTested)
-                    .forEach(serviceVersion -> argumentsList.add(Arguments.of(httpClient, serviceVersion)));
-            });
-        return argumentsList.stream();
+        CertificateServiceVersion[] filteredKeyServiceVersion =
+            Arrays.stream(CertificateServiceVersion.values())
+                .filter(CertificateClientTestBase::shouldServiceVersionBeTested)
+                .toArray(CertificateServiceVersion[]::new);
+
+        boolean rollingServiceVersion = SERVICE_VERSION_FROM_ENV != null && SERVICE_VERSION_FROM_ENV
+            .equalsIgnoreCase(AZURE_TEST_SERVICE_VERSIONS_VALUE_ROLLING);
+        return getArgumentsFromServiceVersion(Arrays.asList(filteredKeyServiceVersion), rollingServiceVersion);
```

Comment:
I would simplify this code further by moving more of the logic into the `getArgumentsFromServiceVersion` method. You could, for example, just pass in the ServiceVersion enum, a predicate (i.e. `CertificateClientTestBase::shouldServiceVersionBeTested`), and that's it. The method could look up the environment variable.

Reply:
The first parameter needs to get filtered first, and need some `getLatest()` API specifically for each client. So leave it there.

We can simplify the second parameter by passing env value to TestBase.